datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -1005,46 +1016,46 @@ async fn collect_left_input(
     let mut offset = 0;
 
     // Updating hashmap starting from the last batch
-    let batches_iter = batches.iter().rev();
-    for batch in batches_iter.clone() {
-        hashes_buffer.clear();
-        hashes_buffer.resize(batch.num_rows(), 0);
-        update_hash(
-            &on_left,
-            batch,
-            &mut *hashmap,
-            offset,
-            &random_state,
-            &mut hashes_buffer,
-            0,
-            true,
-        )?;
-        offset += batch.num_rows();
+
+    let batches_iter = batches.iter();
+    let mut coalescer = BatchCoalescer::new(schema, num_rows);
```

Comment:
I think `BatchCoalescer` has no benefit here as it will currently concatenate to a single batch.